sources/BigInt.swift

```diff
@@ -259,6 +259,17 @@ extension BigInt: IntegerArithmetic {
     public static func %(a: BigInt, b: BigInt) -> BigInt {
         return BigInt(abs: a.abs % b.abs, negative: a.negative)
     }
+  
+    /// Return the result of `a` mod `b`.
+    public static func modulus(a: BigInt, b: BigInt) -> BigInt {
+        if a.negative == b.negative {
+            return BigInt(abs: a.abs % b.abs, negative: a.negative)
+        }
+        else {
+            let floor = ((a / b) - 1) * b
+            return BigInt(abs: (floor.abs - a.abs) % b.abs, negative: b.negative)
```

Comment:
For my use only the cases where `b` is positive were important. So I decided to implement the behavior that matches WolframAlpha for the cases where `b` is negative. But I think you're right: What we want here is probably that the result is always nonnegative (so a `BigUInt`?).
So I absolutely agree with your code which is way more elegant.